module/Activity/src/Activity/Controller/ActivityController.php

```diff
@@ -181,31 +191,48 @@ public function signoffAction()
 
         //Assure the form is valid
         if (!$form->isValid()){
-            $params = $this->viewAction();
-            $params['error'] = $translator->translate('Wrong form');
-            return $params;
+            $message = $translator->translate('Wrong form');
+            $this->redirectActivityRequest($id, false, $message);
+            return;
         }
 
         if (!$signupService->isAllowedToSubscribe()) {
-            $params = $this->viewAction();
-            $params['error'] = $translator->translate('You have to be logged in to subscribe for this activity');
-            return $params;
+            $message = $translator->translate('You have to be logged in to subscribe for this activity');
+            $this->redirectActivityRequest($id, false, $message);
+            return;
         }
 
         $identity = $this->getServiceLocator()->get('user_role');
         $user = $identity->getMember();
 
         if (!$signupService->isSignedUp($activity, $user)) {
-            $params = $this->viewAction();
-            $params['error'] = $translator->translate('You are not subscribed for this activity!');
-            return $params;
+            $message = $translator->translate('You are not subscribed for this activity!');
+            $this->redirectActivityRequest($id, false, $message);
+            return;
         }
 
         $signupService->signOff($activity, $user);
-        $params = $this->viewAction();
-        $params['success'] = true;
+        $message = $translator->translate('Successfully unsubscribed');
+        $this->redirectActivityRequest($id, true, $message);
+    }
 
-        return $params;
+    /**
+     * Redirects to the view of the activity with the given $id, where the
+     * $error message can be displayed if the request was unsuccesful (i.e.
+     * $success was false)
+     *
+     * @param int $id
+     * @param boolean $success Whether the request was successful
+     * @param string $message
+     */
+    protected function redirectActivityRequest($id, $success, $message)
+    {
+        $activityRequestSession = new SessionContainer('activityRequest');
+        $activityRequestSession->success = $success;
```

Comment:
Zend framework actually has a built in function for sending messages http://framework.zend.com/manual/current/en/modules/zend.mvc.plugins.html#zend-mvc-controller-plugins-flashmessenger